### ISSUE TITLE
[BUG] motion.custom errors when used with react-router "Link"

### ISSUE THREAD

**Issue by DanProudfoot** [framer/motion#336]
**Describe the bug**
I'm trying to wrap a react-router Link component with motion.custom. 

```
Uncaught Error: No `ref` found. Ensure components created with `motion.custom` forward refs using `React.forwardRef`
```

As far as I can tell, Link does use forwardRef

**Steps to reproduce**

1. 
```
import { Link } from 'react-router-dom';
import { motion } from 'framer-motion';

const MotionLink = motion.custom(Link);
```
2. Use MotionLink, and everything falls apart

**Expected behavior**
A Link component that accepts motion props to animate

**Comment by DanProudfoot** [comment]
Ok, so this is dumb on my end. Apparently Link didn't actually forwardRef, until about 4 hours after I made this post. 

Coincidentally, I just happened to make this post the exact day this issue was being fixed on react-router's end.

Sigh.

One thing I have noitced though is that all the motion-related props get spread onto the custom motion component. So now i have an <a /> with lots with variants on it
